pkg/telemetry/telemetry.go

```diff
@@ -1,10 +1,25 @@
 package telemetry
 
-import "go.uber.org/zap"
+import (
+	"context"
+
+	"go.opentelemetry.io/contrib/exporters/autoexport"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/propagation"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+	"go.uber.org/zap"
+)
+
+var Resource = resource.NewWithAttributes(
+	semconv.SchemaURL,
+	semconv.ServiceName("glide"),
+)
 
 type Config struct {
 	LogConfig *LogConfig `yaml:"logging" validate:"required"`
-	// TODO: add OTEL config
```

Comment:
Thoughts out loud.

This auto instrumentation follows a bit different idea than Glide's config. 
I would like to see all configuration done via Glide config, so there is one single place to take a look. 
Autoinstrumentation creates another place to look at which environment variables which means that now we would have two places to consider in order to fully understand a gateway setup.

Not sure if that's going to be confusing for end users or not.

Functionally though, autoinstrumentation seems to have everything we need:
- ability to disable/enable trace exporting
- ability to disable enable metric exporting (independently from tracing and vice versa)

https://github.com/EinStack/charts/blob/main/charts/glide/templates/_pod.tpl#L32-L50 - the helm chart is ready to accept additional env vars, too.

Reply:
Personally I would rather stick with the standards. Users configuring telemetry should only care about their otel setup, we can leverage otel documentation meaning we don't need to upgrade telemetry docs whenever we upgrade our otel  packages.


However, one thing I agree with you is the options which are essential for instrumentation. Things like instance id's you mentioned.  Those indeed needs to be set in configuration.

Reply:
> Users configuring telemetry should only care about their otel setup, we can leverage otel documentation meaning we don't need to upgrade telemetry docs whenever we upgrade our otel packages.

I guess it's saving us a bit of typing by providing those registries and implicit configurations via env vars. That's the biggest advantages I see. Also, some configs may appear in the upcoming versions of OTEL SDK that are available right away as we upgrade SDK (without a need to propagate them manually).